libsolidity/ast/Types.h

```diff
@@ -372,6 +373,14 @@ class Type
 	/// are returned without modification.
 	virtual TypeResult interfaceType(bool /*_inLibrary*/) const { return nullptr; }
 
+	/// @returns list of subtypes for composite types (base type for array, members for structs,
+	/// value type for mappings) or empty list for ordinary types
+	virtual std::vector<Type const*> subtypes() const { return {}; }
+	/// @returns list of collision-prone subtypes (with an approximate size exceeding 2**64)
+	std::vector<Type const*> storageOversizedSubtypes() const;
+	/// @returns true if this is valid to be stored in data location _loc
+	BoolResult validForLocation(DataLocation _loc, std::set<std::string>* _seen = nullptr) const;
```

Comment:
```suggestion
	BoolResult validForLocation(DataLocation _loc, std::set<std::string>* io_seen = nullptr) const;
```

This parameter is used for both input and output.

Reply:
OK. Uhh... Is there another `io_` in our codebase?

Reply:
There's one in `CODING_STYLE.md` :P
Well, `o_` seems to be a bit more popular at about 50 uses.
Maybe we should change the coding style then?

But I do think that a prefix for output parameters makes it easier to see that they're used that way. I wish we used it more consistently. Not necessarily `io_` since it's a bit ugly but some convention would be nice.

Reply:
>  Not necessarily `io_` since it's a bit ugly
We may argue that ugliness of the name reflects the ugliness of the input/output parameters ).

Reply:
That's exactly my point :) For me the output parameters are always suspect so I want them to stand out :)